src/stim/circuit/gate_data.h

```diff
@@ -44,7 +44,46 @@ struct ErrorAnalyzer;
 constexpr uint8_t ARG_COUNT_SYGIL_ANY = uint8_t{0xFF};
 constexpr uint8_t ARG_COUNT_SYGIL_ZERO_OR_ONE = uint8_t{0xFE};
 
-inline uint8_t gate_name_to_id(const char *v, size_t n) {
+enum class Gates : uint8_t {
+    // From gate_data_annotations.cc
+    DETECTOR, OBSERVABLE_INCLUDE, TICK, QUBIT_COORDS, SHIFT_COORDS,
+    // Frome gate_data_blocks.cc
+    REPEAT,
+    // From gate_data_collapsing.cc
```

Comment:
Avoid coupling the enum definition to the file structure. Say what they are instead of where they are (e.g. here it's collapsing gates).